v2/src/lib.rs

```diff
@@ -0,0 +1,837 @@
+        };
+
+        let db = get_db_name(self.config.clone()).unwrap();
+
+        // Note: tags are stored as strings in InfluxDB, while fileds are typed.
+        // For simpler/faster deserialization, we store encoding, timestamp and base64 as fields.
+        // while the kind is stored as a tag to be indexed by InfluxDB and have faster queries on it.
+        let zenoh_point = vec![DataPoint::builder(measurement.clone())
+            .tag("kind", "PUT")
+            .tag("ztimestamp", timestamp.to_string())
+            .field(
+                "encoding_prefix",
+                u8::from(*value.encoding.prefix()).to_string(),
+            ) //issue with this data-type, converting it to string for now, will have to check later
+            .field("encoding_suffix", value.encoding.suffix())
+            .field("base64", base64)
+            .field("value", strvalue)
+            .build()
+            .expect("error in building the write query")];
+
+        match async_std::task::block_on(async {
+            self.client.write(&db, stream::iter(zenoh_point)).await
+        }) {
+            Ok(_) => Ok(StorageInsertionResult::Inserted),
+            Err(e) => bail!(
+                "Failed to put Value for {:?} in InfluxDb storage : {}",
+                measurement,
+                e
+            ),
+        }
+    }
+
+    async fn delete(
+        &mut self,
+        key: Option<OwnedKeyExpr>,
+        timestamp: Timestamp,
+    ) -> ZResult<StorageInsertionResult> {
+        let measurement = key.unwrap_or_else(|| OwnedKeyExpr::from_str(NONE_KEY).unwrap());
+        // Note: assume that uhlc timestamp was generated by a clock using UNIX_EPOCH (that's the case by default)
+        // delete all points from the measurement that are older than this DELETE message
+        // (in case more recent PUT have been recevived un-ordered)
+        // let query = InfluxRQuery::new(format!(
+        //     r#"DELETE FROM "{}" WHERE time < {}"#,
+        //     measurement, influx_time
+        // ));
+        let db = get_db_name(self.config.clone()).unwrap();
+
+        let start = NaiveDateTime::UNIX_EPOCH;
+        //  let deletion_time =        NaiveDateTime::from_timestamp_millis(timestamp.get_time().0.try_into().unwrap()).unwrap();
+        let curr_deletion_time = Utc::now().naive_utc();
+        let predicate = None; //can be specified as tag or field values
+        debug!("Delete {:?} with Influx query", measurement);
+
+        if let Err(e) = self
+            .client
+            .delete(&db, start, curr_deletion_time, predicate)
+            .await
+        {
+            bail!(
+                "Failed to delete points for measurement '{}' from InfluxDb storage : {}",
+                measurement,
+                e
+            )
+        }
+        // store a point (with timestamp) with "delete" tag, thus we don't re-introduce an older point later;
+        // filling fields with dummy values
```

Comment:
In Influxdb2 `get()` will throw a error if we do not have same field and tag for each entry in the db since it serializes all values to same type. Adding the same change to v1 for compatibility.